src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -298,6 +298,9 @@ public int Run()
 
             var logger = new Logger(Console.Out, ilProvider, Get(_command.IsVerbose), ProcessWarningCodes(Get(_command.SuppressedWarnings)),
                 Get(_command.SingleWarn), Get(_command.SingleWarnEnabledAssemblies), Get(_command.SingleWarnDisabledAssemblies), suppressedWarningCategories);
+
+            ilProvider = new FeatureSwitchManager(ilProvider, logger, featureSwitches);
```

Comment:
Could you also move the `List<KeyValuePair<string, bool>> featureSwitches` block that was right above it so that feature switch handling is not spread around the method?